ang/civicase/case/details/directives/case-details.directive.js

```diff
@@ -82,12 +82,18 @@
      * Opens the popup for Creating Email
      */
     $scope.createEmail = function () {
+      var caseClients = _.filter($scope.item.contacts, function (contact) {
```

Comment:
This logic is same as above, we should move the logic in a common service called `getCaseClientsIds`.

Reply:
@deb1990 : If you still feel they are the same logic we can have a discussion. Please let me know

Reply:
I just checked, and yes its still the same logic

```javascript
_.chain(activity['case_id.contacts'])
	.filter(function (contact) {
  		return contact.role === 'Client';
	})
	.map(function (client) {
  		return client.contact_id;
	})
	.value();
```